kong/tools/queue.lua

```diff
@@ -0,0 +1,408 @@
+-- Queue with retryable processing
+--
+-- This is a queue of "entries". Entries can be any Lua value.
+--
+-- The queue has two mandatory parameters:
+--
+-- `name` is the name of the queue. It uniquely identifies the queue across the
+-- system.  If two plugin instances use the same queue name, they will share one
+-- queue and their queue related configuration must match.
```

Comment:
You need to put this in the user facing document and on the field schema description (LUa comment) in queue_schema.lua as well. It is not going to be obvious to the user.